src/main/java/be/syntra/devshop/DevshopFront/services/CartServiceImpl.java

```diff
@@ -122,12 +147,33 @@ private void setCartToFinalized(CartDto cartDto) {
 
     @Override
     public BigDecimal getCartTotalPrice(CartDto currentCart) {
-        return currentCart.getProducts().stream()
-                .map(product -> {
-                    BigDecimal pricePerProduct = product.getPrice();
-                    int totalInCart = product.getTotalInCart();
-                    return pricePerProduct.multiply(new BigDecimal(totalInCart));
+        return currentCart.getCartProductDtoList().stream()
+                .map(cartProductDto -> {
+                    BigDecimal productPrice = cartProductDto.getProductDto().getPrice();
+                    int count = cartProductDto.getCount();
+                    return productPrice.multiply(new BigDecimal(count));
                 })
                 .reduce(BigDecimal.ZERO, BigDecimal::add);
     }
+
+    @Override
+    public CartDisplayDto getCartDisplayDto() {
+        return CartDisplayDto.builder().cartDisplayProductDtoList(
```

Comment:
really torn now. On one hand, love the incorporation of streams and one-lining everything. On the other hand. I'm so terribly confused with so many Carts and Dto's.... Would you agree to extract the streams to seperate methods? And just keep the inlined builder?
Perhaps I've said it before, but I really don't like 'following the pattern or convention' when it means nesting objects like this....

Reply:
i understand, i'll have to force myself to look again at code that i write and looks 'weird' like this one, so now i extracted the methods and tried to use sensible naming.